espnet/nets/pytorch_backend/e2e_asr_transformer.py

```diff
@@ -634,3 +634,74 @@ def calculate_all_attentions(self, xs_pad, ilens, ys_pad):
                 ret[name + "_time"] = m.attn_t.cpu().numpy()
                 ret[name + "_freq"] = m.attn_f.cpu().numpy()
         return ret
+
+    def ctc_align(self, x, y):
+        """E2E CTC alignment.
+
+        :param torch.Tensor x: input acoustic feature (T, D)
+        :param torch.Tensor y: id sequence tensor (L)
+        :return: best alignment results
+        :rtype: list
+        """
+
+        def interpolate_blank(label, blank_id=0):
+            """Insert blank token between every two label token."""
+            label = numpy.expand_dims(label, 1)
+            blanks = numpy.zeros((label.shape[0], 1), dtype=numpy.int64) + blank_id
+            label = numpy.concatenate([blanks, label], axis=1)
+            label = label.reshape(-1)
+            label = numpy.append(label, label[0])
+            return label
+
+        enc_output = self.encode(x).unsqueeze(0)
+        lpz = self.ctc.log_softmax(enc_output)
+        lpz = lpz.squeeze(0)
+
+        y_int = interpolate_blank(y, self.blank)
+
+        logdelta = (
+            numpy.zeros((lpz.size(0), len(y_int))) - 100000000000.0
+        )  # log of zero
+        state_path = (
+            numpy.zeros((lpz.size(0), len(y_int)), dtype=numpy.int16) - 1
+        )  # state path
+
+        logdelta[0, 0] = lpz[0][y_int[0]]
+        logdelta[0, 1] = lpz[0][y_int[1]]
+
+        import six
```

Comment:
How about moving this "import" to the beginning of the file?

Reply:
Thanks for the suggestion.